src/browser/template-api.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2018-2021 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import { AxiosInstance } from 'axios';
+import { IDevWorkspaceTemplate, IDevWorkspaceTemplateApi } from '../types';
+import { devworkspaceVersion, devworkspaceGroup, devworkspaceTemplateSubresource } from '../common';
+
+export class RestDevWorkspaceTemplateApi implements IDevWorkspaceTemplateApi {
+  private axios: AxiosInstance;
+
+  constructor(axios: AxiosInstance) {
+    this.axios = axios;
+  }
+
+  async getAllTemplates(namespace: string): Promise<IDevWorkspaceTemplate[]> {
+    const resp = await this.axios.get(
+      `/apis/${devworkspaceGroup}/${devworkspaceVersion}/namespaces/${namespace}/${devworkspaceTemplateSubresource}`
+    );
+    return resp.data.items;
+  }
+
+  async getTemplateByName(
```

Comment:
create and delete are without Template, let's align their names, like listInNamespace, getByName, create, delete.
WDYT?